server/routes/paraRouter.ts

```diff
@@ -0,0 +1,171 @@
+ * permissions and limitations under the License.
+ */
+
+import { schema } from '@kbn/config-schema';
+import { IRouter, IKibanaResponse, ResponseError } from '../../../../src/core/server';
+import { API_PREFIX, wreckOptions } from '../../common';
+import BACKEND from '../adaptors';
+
+export function ParaRouter(router: IRouter) {
+  // Updates the input content in a paragraph --> Runs the paragraph --> Fetches the updated Paragraph (with new input content)
+  router.post(
+    {
+      path: `${API_PREFIX}/para/update/run/`,
+      validate: {
+        body: schema.object({
+          noteId: schema.string(),
+          paraId: schema.string(),
+          paraInp: schema.string(),
+        }),
+      },
+    },
+    async (context, request, response): Promise<IKibanaResponse<any | ResponseError>> => {
+      let runResponse = {};
+      try {
+        runResponse = await BACKEND.updateRunPara(context, request.body, wreckOptions);
+        return response.ok({
+          body: runResponse,
+        });
+      } catch (error) {
+        return response.custom({
+          statusCode: error.statusCode || 500,
+          body: {
+            body: runResponse,
+          },
+        });
+      }
+    }
+  );
+
+  // Updates the input content in a paragraph --> Fetches the updated Paragraph (with new input content)
+  router.put(
+    {
+      path: `${API_PREFIX}/para/`,
+      validate: {
+        body: schema.object({
+          noteId: schema.string(),
+          paraId: schema.string(),
+          paraInp: schema.string(),
+        }),
+      },
+    },
+    async (context, request, response): Promise<IKibanaResponse<any | ResponseError>> => {
+      let saveResponse = {};
+      try {
+        saveResponse = await BACKEND.updateFetchPara(context, request.body, wreckOptions);
+        return response.ok({
+          body: saveResponse,
+        });
+      } catch (error) {
+        return response.custom({
+          statusCode: error.statusCode || 500,
+          body: {
+            body: saveResponse,
+          },
+        });
+      }
+    }
+  );
+
+  // Add a new para
+  router.post(
+    {
+      path: `${API_PREFIX}/para/`,
+      validate: {
+        body: schema.object({
+          noteId: schema.string(),
+          paraIndex: schema.number(),
+          paraInp: schema.string(),
+        }),
+      },
+    },
+    async (context, request, response): Promise<IKibanaResponse<any | ResponseError>> => {
+      let addResponse = {};
+      try {
+        addResponse = await BACKEND.addNewPara(context, request.body, wreckOptions);
+        return response.ok({
+          body: addResponse,
+        });
+      } catch (error) {
+        return response.custom({
+          statusCode: error.statusCode || 500,
+          body: {
+            body: addResponse,
+          },
+        });
+      }
+    }
+  );
+
+  // Delete a paragraph
+  router.delete(
+    {
+      path: `${API_PREFIX}/para/{ids*2}`,
```

Comment:
can you explain this `{ids*2}`? I've never seen this syntax in an API route

Reply:
From Hapi Docs: https://hapi.dev/tutorials/routing/?lang=en_US#-multi-segment-parameters